lib/pub_cache.dart

```diff
@@ -52,10 +49,30 @@ class PubCache {
     return dir.existsSync() ? dir.listSync() : [];
   }
 
+  /**
+   * Return applications that have been installed via `pub global activate`.
+   */
   List<Application> getGlobalApplications() => _applications;
 
+  /**
+   * Get all the packages and their versions that have been installed into the
+   * cache.
+   */
   List<PackageRef> getPackageRefs() => _packageRefs;
 
+  /**
+   * Return the list of package names (not versions) that area available in the
+   * cache.
+   */
+  List<String> getCachedPackages() =>
+      new Set.from(getPackageRefs().map((p) => p.name)).toList();
```

Comment:
Hilarious.  I'm guessing this won't get called more than once?  If that turns out not to be true, it may make sense to calculate just once and store it?